compiler/rustc_passes/src/intrinsicck.rs

```diff
@@ -255,10 +255,18 @@ impl ExprVisitor<'tcx> {
         // register class.
         let asm_arch = self.tcx.sess.asm_arch.unwrap();
         let reg_class = reg.reg_class();
-        let supported_tys = reg_class.supported_types(asm_arch);
-        let feature = match supported_tys.iter().find(|&&(t, _)| t == asm_ty) {
-            Some((_, feature)) => feature,
-            None => {
+        let found_supported_ty = match reg_class.supported_types(asm_arch) {
+            // FIXME(eddyb) consider skipping the "cannot use value of type" error
+            // above, letting the codegen backend handle non-scalar/vector types.
```

Comment:
We need to do this as part of the type-checking passes since we want to validate these before monomorphization.

Reply:
I meant specifically for the `Any` case. AFAIK SPIR-V should be able to handle any `Sized` value (similar to LLVM "first-class aggregrates" but even more first-class).